src/Model/Profile.php

```diff
@@ -88,9 +90,15 @@ public static function formatLocation(array $profile)
 	 * @param int     $profile      int
 	 * @param array   $profiledata  array
 	 * @param boolean $show_connect Show connect link
+	 * @throws InvalidArgumentException Thrown when a parameter is invalid
 	 */
-	public static function load(App $a, $nickname, $profile = 0, $profiledata = [], $show_connect = true)
+	public static function load(App $a, $nickname, $profile = 0, array $profiledata = [], $show_connect = true)
 	{
+		if (empty($nickname)) {
```

Comment:
Why should there be some exception for just some empty nickname? This isn't the end of the universe. An exception should - from my point of view - mostly be used at some critical part like where data is processed that would otherwise be corrupted.

Reply:
The name of the exception answers your question: `InvalidArgumentException`, an argument (aka. parameter) of a method is invalid as it contains an unexpected value.

Otherwise corrupted data should be stopped with `assert()` not exceptions as this is a flaw (in perspective of code view) in code logic. An exception is (as the word says) is something harder, like (under Java) an `IOExcecption` where some I/O operation went wrong because of write-protected or full disk or corrupted sectors. This is some "higher" power (not logic's fault) as just a flaw in logic.

Reply:
For me this sounds like blowing up your car with TNT because a tail light has an issue. This exception would possibly be shown to the page visitor. But there is one rule: Never give processing information to some user. Only the admin should know if there had been a problem. And then some logging really is okay.

Reply:
Exception should never make it up to the user. This would indicate a bad mistake is still present in code and was not located. You never want to let a car run out from factory to the happy brand-new-car-owning driver when you have not properly tested it. And exceptions can be nicely tested with unit tests (what you also will do with a car).

Reply:
If you want to keep the exception, you have to wrap every `Profile::load` in a `try {} catch ()` block.

Reply:
That would use the exception as flow-control mechanism which you should never do. Exceptions are no replacement for proper `if()` and `return` statements. :-) And especially this exception, you **never** want to catch it.

Reply:
Except if we leave the Exception as is, if it's ever thrown, it's going to crash the whole page for no good reason while at best outputting a cryptic error message for all the node's users.

Exceptions are a flow control mechanism and should be handled one way or another, even if it's for displaying a generic error message and writing the error message to the log or sending an email to the admins.

What would be their use otherwise?

Reply:
Here's how I'm using Exceptions in an environment ready to handle them:
```php
/**
 * @param type $param
 * @return expected_type
 */
function method()
{
    if (empty($param)) {
        throw new BadMethodCallException(...);
    }

    if (!is_type($param)) {
        throw new InvalidParameterException(...);
    }

    $expected_type = something($param);

    return $expected_type;
}
```

This ensures that the return value is always from the expected type because we interrupt the function beforehand if there's something wrong with the parameter. However, it isn't always critical to the application, and catching the Exception allows to keep displaying a page (which may be an error page) with the assurance that we didn't get funky data from `method()`.

Reply:
This part is okay for me as well because it makes sure that parameters/arguments are always valid. For the user, if such thing happens, let's output the infamous 500 error message, maybe?

Reply:
Definitely, but you have to add it in this PR if you plan on keeping the Exception. Typically. you would put the `try {} catch ()` block in `index.php` wrapping the module function calls (*_init, *_content, etc...).

Reply:
But in this case, when there is no nickname found, I don't see a reason to throw out a "500".

Reply:
If `$nickname` is empty, it means a bad logic error as nicknames cannot be empty (Example: Hello ` `). I think this is a general change to Friendica (not opposing my proposed change) and it can be discussed endless. If we really want this here fixed, I add `try {} catch {}` to `index.php` with a "catch-all catch" which will result in a proper error message as something really bad has happened (if `$nickname` is empty, other code has failed to stop this from further processing).

We are not getting this PR through if we continue to argue what is bad or good practice.